rs/embedders/src/wasmtime_embedder.rs

```diff
@@ -226,17 +219,8 @@ impl WasmtimeEmbedder {
     pub fn wasmtime_execution_config(embedder_config: &EmbeddersConfig) -> wasmtime::Config {
         let mut config = wasmtime_validation_config(embedder_config);
 
-        // Wasmtime features that differ between Wasm validation and execution.
-        // Currently these are multi-memories and the 64-bit memory needed for
-        // the Wasm-native stable memory implementation.
-        if embedder_config.feature_flags.write_barrier == FlagStatus::Enabled
-            || embedder_config.feature_flags.wasm_native_stable_memory == FlagStatus::Enabled
-        {
-            config.wasm_multi_memory(true);
-        }
```

Comment:
Since this can be a bit unintuitive at first: dropping the `wasm_native_stable_memory` flag means that we should then unconditionally use multi memory configuration irrespective of the write barrier flag. Otherwise, we miss some cases. Initially I kept the condition on the write barrier and some things were not working.